comp/core/autodiscovery/autodiscoveryimpl/autoconfig_test.go

```diff
@@ -384,9 +384,9 @@ func TestResolveTemplate(t *testing.T) {
 	}
 	// there are no template vars but it's ok
 	ac.processNewService(ctx, &service) // processNewService applies changes
-	clk := clock.NewMock()
-	clk.Add(2000 * time.Millisecond)
-	assert.Equal(t, sch.scheduledSize(), 1)
+	testutil.AssertTrueBeforeTimeout(t, 10*time.Millisecond, 4*time.Second, func() bool {
```

Comment:
Should we use testify's `assert.Eventually` instead of our own utils?
I know that a few tests use `testutil` but it was not a dependency of this file and I think we should probably migrate to `assert` instead of using our own.